server/src/main/java/org/opensearch/index/query/ExistsQueryBuilder.java

```diff
@@ -202,13 +202,13 @@ private static Query newObjectFieldExistsQuery(QueryShardContext context, String
         BooleanQuery.Builder booleanQuery = new BooleanQuery.Builder();
         Collection<String> fields = context.simpleMatchToIndexNames(objField + ".*");
         for (String field : fields) {
-            int dotPos = field.lastIndexOf('.');
-            if (dotPos > 0 && field.charAt(dotPos + 1) == '_') {
-                // This is a subfield (e.g. prefix) of a complex field type. Skip it.
-                continue;
+            try {
+                Query existsQuery = context.getMapperService().fieldType(field).existsQuery(context);
+                booleanQuery.add(existsQuery, Occur.SHOULD);
+            } catch (Exception e) {
```

Comment:
Am thinking rather than adding this catch all we can check if the field type is `DerivedFieldType` here instead and leave the subfield check above?

```java
            MappedFieldType mappedFieldType = context.getMapperService().fieldType(field);
            if (mappedFieldType instanceof DerivedFieldType) continue;
```

Reply:
I see, In that case I like your suggestion of catching UOE specifically over the Exception catch all.